src/context/TodoContext.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {
+  MutableRefObject,
+  ReactNode,
+  SetStateAction,
+  createContext, useContext, useEffect, useRef, useState,
+} from 'react';
+import { editTodo, getTodos } from '../api/todos';
+import { SortTypes, Todo } from '../types/Todo';
+
+export interface TContext {
+  todos: Todo[];
+  setTodos: React.Dispatch<SetStateAction<Todo[]>>;
+  hasError: null | string;
+  handleError: (error: string) => void;
```

Comment:
Please see the comment related to types in `const [hasError, setHasError]`.
Instead of `string` let's use a string union